### DISCUSSION TITLE
Is the " || "required when signcrypting?

### DISCUSSION THREAD

**Discussion by superpoincare** [jedisct1/libsodium#1187] [Q&A] [answered]
According to the libsodium documentation, 

`Finally, if, for some reason, you want to implement your own signcryption scheme:
If public verifiability is not required, sign (encryption_key || message) first then encrypt (recipient_id || signature || message).

…
`

I am curious if the vertical line characters and spaces are required, or is it written like that for ease of reading, because the sign function doesn't add the " || " itself.

**Comment by enkore** [comment, marked as answer]
|| means concatenation (often with implied length encoding if variable length)

**Comment by superpoincare** [comment]
Thank you.
